packages/webapp/webapp_server.js

```diff
@@ -812,22 +804,82 @@ var runWebAppServer = function () {
+    var startHttpServer = function(listenOptions) {
+      httpServer.listen(listenOptions, Meteor.bindEnvironment(function() {
+        if (process.env.METEOR_PRINT_ON_LISTEN)
+          console.log("LISTENING"); // must match run-app.js
+
+        var callbacks = onListeningCallbacks;
+        onListeningCallbacks = null;
+        _.each(callbacks, function (x) { x(); });
+
+      }, function (e) {
+        console.error("Error listening:", e);
+        console.error(e && e.stack);
+      }));
+    };
+
+    localPort = isNaN(Number(localPort)) ? localPort : Number(localPort);
 
-    }, function (e) {
-      console.error("Error listening:", e);
-      console.error(e && e.stack);
-    }));
+    if (typeof localPort == "string") {
+      var socketPath = localPort;
+      var listenOptions = { path: socketPath };
 
+      // Test to see if the socket path is Windows Server-style named pipe with the
+      // format: \\.\pipe\{randomstring} or \\{servername}\pipe\{randomstring}
+      // Otherwise, treat it as a UNIX domain socket path pointing to a socket file
+      if ( /\\\\?.+\\pipe\\?.+/.test(socketPath)) {
+        startHttpServer(listenOptions);
+      } else {
+        if (fs.existsSync(socketPath)) {
+          if (fs.statSync(socketPath).isSocket()) {
+            var clientSocket = new net.Socket();
+            clientSocket.on('error', Meteor.bindEnvironment(function(e) {
+              //If there is an existing socket file that we cannot connect to,
+              //it must be stale as it is not associated with a running server
+              if (e.code == 'ECONNREFUSED') {
+                console.log("Deleting stale socket file");
+                fs.unlinkSync(socketPath);
+                startHttpServer(listenOptions);
+              }
+            }));
+            clientSocket.connect({ path: socketPath }, function() {
+              //If there is a running server listening on that socket file,
+              //we have to leave the file alone to avoid preventing other clients
+              //from connecting to it.
+              console.error("Another server is already listening on socket: " + socketPath + ", exiting.");
+              process.exit();
+            });
+          } else {
+            //There is an existing file at socketPath that is not a socket,
+            //so we will leave it alone and exit
+            console.error("Cannot listen on socket: " + socketPath + ", exiting.");
+            process.exit();
+          }
+        } else {
+          startHttpServer(listenOptions);
+        }
+        //Clean up the socket file when we've finished to avoid leaving a stale one
+        //That situation should only be able to happen if the running node process is
+        //killed via signal 9 (SIGKILL).
+        process.on('exit', Meteor.bindEnvironment(function(e) {
+          fs.unlinkSync(socketPath);
+        }));
+        process.on('SIGINT', Meteor.bindEnvironment(function(e) {
```

Comment:
Without this cleanup code, a stale UNIX socket file (.sock) file is left on disk if you terminate the node process with Control-C (SIGINT).